### PR TITLE
Align Dependabot coverage and add the dev-fast build fragment

### PR DESCRIPTION
**Pull request by leynos**
## Summary

This branch applies Wave 1 of the Rust estate baseline remediation
(Operation Parabellum, phase 2). It aligns
[.github/dependabot.yml](https://github.com/leynos/rstest-bdd/blob/parabellum-wave-1/.github/dependabot.yml)
with the canonical Dependabot reference: one update stanza per package
ecosystem the repository uses (`cargo`, `rust-toolchain`, `uv`, `github-actions`), each stanza labelled
`dependencies` plus its channel label, and GitHub Actions updates batched
into a single pull request via a wildcard group. It also adds the opt-in dev-fast build
fragment with `dev-build` and `dev-test` Make targets, signposted in
[AGENTS.md](https://github.com/leynos/rstest-bdd/blob/parabellum-wave-1/AGENTS.md).

## Review walkthrough

- Start with
  [.github/dependabot.yml](https://github.com/leynos/rstest-bdd/blob/parabellum-wave-1/.github/dependabot.yml)
  for the ecosystem coverage and labels.
- Then review
  [tools/dev-fast/config.toml](https://github.com/leynos/rstest-bdd/blob/parabellum-wave-1/tools/dev-fast/config.toml)
  — a copy of the canonical fragment — and the `dev-build`/`dev-test`
  targets appended to the
  [Makefile](https://github.com/leynos/rstest-bdd/blob/parabellum-wave-1/Makefile).
- Finish with the signposting section appended to
  [AGENTS.md](https://github.com/leynos/rstest-bdd/blob/parabellum-wave-1/AGENTS.md).

## Validation

- Audited the amended tree with the Concordat rule packages
  `dependabot-baseline` and `rust-dev-fast-baseline`: `dependabot-baseline` compliant, `rust-dev-fast-baseline` compliant.

## Notes

- The canonical labels (including the Dependabot channel labels this
  configuration references) were created on the repository ahead of this
  pull request, so no stanza names a missing label.
- The dev-fast fragment is strictly opt-in local tooling: it is only
  applied when passed explicitly with `--config`, so continuous
  integration, release, coverage, and verification builds are untouched.
  The targets need a nightly toolchain and, on Linux, a `mold` binary on
  the `PATH`; repositories still pinned to stable gain the wiring now and
  the capability when Wave 2 moves the pin to nightly.

## Summary by Sourcery

Align dependency automation, add opt-in fast development builds, and improve cross-platform test diagnostics.

New Features:
- Add opt-in accelerated development build and test targets using the Cranelift backend and platform-appropriate linker configuration.
- Expand Dependabot coverage to include uv and rust-toolchain updates with ecosystem-specific labels and scheduling.

Bug Fixes:
- Improve Windows and LSP smoke-test failure diagnostics by preserving Cargo output and capturing bounded language-server stderr.
- Ensure LSP smoke tests perform bounded, clean process shutdown and report exit failures clearly.

Enhancements:
- Normalize example workspace lint inheritance and add contract coverage for the dev-fast Makefile configuration.
- Update the Lading dependency reference used by local and CI workflows.

CI:
- Update the CI Lading revision used for pre-flight diagnostics.

Documentation:
- Document the opt-in fast development build workflow, prerequisites, configuration, and safety boundaries.
- Add a debugging plan for diagnosing Windows LSP smoke-test pre-flight failures.

Tests:
- Add contract tests covering dev-fast Make targets, Cargo configuration isolation, cross-platform executable paths, and example workspace lint inheritance.
- Add regression coverage for capturing stderr from early-exiting LSP child processes.

Chores:
- Extend spelling exceptions for Cranelift and mold terminology.

## References

- [Lody session](https://lody.ai/leynos/sessions/acdae2a9-5597-4fab-9f0f-a5db83fcee82)

### CODE REVIEW

**Comment by leynos** [comment]
The spelling gate failed on this branch because the appended dev-fast text names the `mold` linker, which the en-GB-oxendict dictionary read as a misspelling of “mould”. The follow-up commit accepts the word via the estate's correction-to-itself pattern in `typos.local.toml` (mirrored in the generated dictionary); the AGENTS.md and Makefile hunks are untouched, so the companion `parabellum-wave-2-3` branches still merge cleanly where they exist.

File: AGENTS.md [hunk 1]
```diff
@@ -384,3 +384,12 @@ The following tooling is available in this environment:
 
 These practices help maintain a high-quality codebase and facilitate
 collaboration.
+
+## Fast development builds
+
+`make dev-build` and `make dev-test` compile with the opt-in Cranelift
+backend and the mold linker configured in `tools/dev-fast/config.toml`.
+They require a nightly toolchain and, on Linux, a `mold` binary on the
+`PATH`. The fragment is passed explicitly with `--config`, so release,
```
[the remark this answers is not in the document]
Comment (leynos) [reply AGENTS.md]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

### ❌ Failed checks (1 error, 2 warnings)

| Check name | Status | Explanation | Resolution |
| :---: | :--- | :--- | :--- |
| Testing (Overall) | ❌ Error | The pull request adds behavioural Makefile functionality without substantive tests. The diff adds `dev-build` and `dev-test`, which invoke Cargo with `tools/dev-fast/config.toml`; the configuration se… | Add substantive integration tests for the new Make targets and configuration. Use a controlled fake `cargo` executable or an isolated fixture so the tests verify that both targets pass the configured path and the expected Cargo subcommands.… |
| Developer Documentation | ⚠️ Warning | The PR adds `make dev-build` and `make dev-test`, the `tools/dev-fast/config.toml` Cargo configuration, the Cranelift backend, and the Linux `mold` prerequisite. The guide requires a nightly toolchain… | Add a section to `docs/developers-guide.md` that documents `make dev-build` and `make dev-test`, `DEV_FAST_CONFIG`, the explicit `--config` usage, the nightly toolchain requirement, the Linux `mold`/`PATH` requirement, Cranelift's debug-bui… |
| Testing (Unit And Behavioural) | ⚠️ Warning | The pull request adds two externally observable command-line workflows, `make dev-build` and `make dev-test`, but adds no unit, behavioural, or end-to-end tests. The diff contains only configuration, … | Add end-to-end integration tests at the Makefile command boundary. Run both targets with the supported nightly toolchain and Linux `mold` setup, and assert that the explicit configuration enables Cranelift and the Linux linker flag. Cover c… |

**Comment by leynos** [comment]
@coderabbitai Please investigate the cause of the following issue using codegraph exploration and research, identify a fix and provide an AI coding agent prompt for the fix:

~~~
     Running tests\smoke_lsp\main.rs (C:\Users\RUNNER~1\AppData\Local\Temp\lading-preflight-target-bfco279r\debug\deps\smoke_lsp-d3da32810dcfa5a5.exe)
error: test failed, to rerun pass `-p rstest-bdd-server --test smoke_lsp`
Unexpected error: Pre-flight cargo test failed with exit code 101: warning: examples\tokio-reminders\Cargo.toml: unused manifest key: package.lints
warning: `tokio-reminders` (manifest) generated 1 warning
warning: examples\todo-cli\Cargo.toml: unused manifest key: package.lints
warning: `todo-cli` (manifest) generated 1 warning
warning: examples\gpui-counter\Cargo.toml: unused manifest key: package.lints
warning: `gpui-counter` (manifest) generated 1 warning
warning: examples\japanese-ledger\Cargo.toml: unused manifest key: package.lints
warning: `japanese-ledger` (manifest) generated 1 warning
~~~

**Comment by leynos** [comment]
@coderabbitai Please investigate the cause of the following issue using codegraph exploration and research, identify a fix and provide an AI coding agent prompt for the fix:

~~~
  
  ================================== FAILURES ===================================
  _______________ test_dev_build_uses_the_dev_fast_cargo_contract _______________
  
  tmp_path = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_build_uses_the_dev_fa0')
  monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x00000232C6FC1B20>
  
      def test_dev_build_uses_the_dev_fast_cargo_contract(
          tmp_path: Path, monkeypatch: pytest.MonkeyPatch
      ) -> None:
          """The build target selects the pinned nightly and explicit fragment."""
          invocation = invoke_dev_fast_target("dev-build", tmp_path, monkeypatch)
      
  >       assert_cargo_invocation(
              invocation=invocation,
              expected_subcommand="build",
          )
  
  scripts\tests\test_dev_fast_makefile_contract.py:140: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  
  invocation = DevFastInvocation(fake_cargo=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_...nd\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n'), real_cargo='C:\\Users\\runneradmin\\.cargo\\bin\\cargo.EXE')
  expected_subcommand = 'build'
  
      def assert_cargo_invocation(
          *,
          invocation: DevFastInvocation,
          expected_subcommand: str,
      ) -> None:
          """Assert that a dev-fast target delegates only to the recording Cargo."""
  >       assert invocation.result.returncode == 0, invocation.result.stdout
  E       AssertionError: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_dev_build_uses_the_dev_fa0\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build
  E         /usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found
  E         make.EXE: *** [Makefile:226: dev-build] Error 127
  E         
  E       assert 2 == 0
  E        +  where 2 = CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-build'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_build_uses_the_dev_fa0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n').returncode
  E        +    where CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-build'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_build_uses_the_dev_fa0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n') = DevFastInvocation(fake_cargo=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_build_uses_the_dev_fa0/cargo'), invocation_log=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_build_uses_the_dev_fa0/cargo-invocations.log'), result=CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-build'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_build_uses_the_dev_fa0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n'), real_cargo='C:\\Users\\runneradmin\\.cargo\\bin\\cargo.EXE').result
  
  scripts\tests\test_dev_fast_makefile_contract.py:115: AssertionError
  _______________ test_dev_test_uses_the_dev_fast_cargo_contract ________________
  
  tmp_path = WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_test_uses_the_dev_fas0')
  monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x00000232C7446990>
  
      def test_dev_test_uses_the_dev_fast_cargo_contract(
          tmp_path: Path, monkeypatch: pytest.MonkeyPatch
      ) -> None:
          """The test target selects the pinned nightly and explicit fragment."""
          invocation = invoke_dev_fast_target("dev-test", tmp_path, monkeypatch)
      
  >       assert_cargo_invocation(
              invocation=invocation,
              expected_subcommand="test",
          )
  
  scripts\tests\test_dev_fast_makefile_contract.py:152: 
  _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
  
  invocation = DevFastInvocation(fake_cargo=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_...und\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n'), real_cargo='C:\\Users\\runneradmin\\.cargo\\bin\\cargo.EXE')
  expected_subcommand = 'test'
  
      def assert_cargo_invocation(
          *,
          invocation: DevFastInvocation,
          expected_subcommand: str,
      ) -> None:
          """Assert that a dev-fast target delegates only to the recording Cargo."""
  >       assert invocation.result.returncode == 0, invocation.result.stdout
  E       AssertionError: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_dev_test_uses_the_dev_fas0\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test
  E         /usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found
  E         make.EXE: *** [Makefile:229: dev-test] Error 127
  E         
  E       assert 2 == 0
  E        +  where 2 = CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-test'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_test_uses_the_dev_fas0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n').returncode
  E        +    where CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-test'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_test_uses_the_dev_fas0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n') = DevFastInvocation(fake_cargo=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_test_uses_the_dev_fas0/cargo'), invocation_log=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_test_uses_the_dev_fas0/cargo-invocations.log'), result=CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-test'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_test_uses_the_dev_fas0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n'), real_cargo='C:\\Users\\runneradmin\\.cargo\\bin\\cargo.EXE').result
  
  scripts\tests\test_dev_fast_makefile_contract.py:115: AssertionError
  ============================== warnings summary ===============================
  .venv-coverage\Lib\site-packages\_pytest\config\__init__.py:1345
    D:\a\rstest-bdd\rstest-bdd\.venv-coverage\Lib\site-packages\_pytest\config\__init__.py:1345: PytestAssertRewriteWarning: Module already imported so cannot be rewritten; slipcover
      self._mark_plugins_for_rewrite(hook, disable_autoload)
  
  -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
  =========================== short test summary info ===========================
  FAILED scripts/tests/test_dev_fast_makefile_contract.py::test_dev_build_uses_the_dev_fast_cargo_contract - AssertionError: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_dev_build_uses_the_dev_fa0\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build
    /usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found
    make.EXE: *** [Makefile:226: dev-build] Error 127
    
  assert 2 == 0
   +  where 2 = CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-build'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_build_uses_the_dev_fa0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n').returncode
   +    where CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-build'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_build_uses_the_dev_fa0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n') = DevFastInvocation(fake_cargo=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_build_uses_the_dev_fa0/cargo'), invocation_log=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_build_uses_the_dev_fa0/cargo-invocations.log'), result=CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-build'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_build_uses_the_dev_fa0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" build\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_build_uses_the_dev_fa0cargo: command not found\nmake.EXE: *** [Makefile:226: dev-build] Error 127\n'), real_cargo='C:\\Users\\runneradmin\\.cargo\\bin\\cargo.EXE').result
  FAILED scripts/tests/test_dev_fast_makefile_contract.py::test_dev_test_uses_the_dev_fast_cargo_contract - AssertionError: C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-0\test_dev_test_uses_the_dev_fas0\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test
    /usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found
    make.EXE: *** [Makefile:229: dev-test] Error 127
    
  assert 2 == 0
   +  where 2 = CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-test'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_test_uses_the_dev_fas0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n').returncode
   +    where CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-test'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_test_uses_the_dev_fas0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n') = DevFastInvocation(fake_cargo=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_test_uses_the_dev_fas0/cargo'), invocation_log=WindowsPath('C:/Users/runneradmin/AppData/Local/Temp/pytest-of-runneradmin/pytest-0/test_dev_test_uses_the_dev_fas0/cargo-invocations.log'), result=CompletedProcess(args=['C:\\mingw64\\bin\\make.EXE', '--no-print-directory', 'dev-test'], returncode=2, stdout='C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-0\\test_dev_test_uses_the_dev_fas0\\cargo +nightly-2026-08-16 --config "tools/dev-fast/config.toml" test\n/usr/bin/sh: line 1: C:UsersrunneradminAppDataLocalTemppytest-of-runneradminpytest-0test_dev_test_uses_the_dev_fas0cargo: command not found\nmake.EXE: *** [Makefile:229: dev-test] Error 127\n'), real_cargo='C:\\Users\\runneradmin\\.cargo\\bin\\cargo.EXE').result
  ============= 2 failed, 79 passed, 3 skipped, 1 warning in 4.93s ==============
~~~

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response. Ensure that this is validated against the *current* version of the codegraph.

If further refinement to address this finding would be deleterious, please supply a clear explanatory one to two paragraph markdown message in a code block that I can paste into the CodeScene web ui's diagnostic suppression function so this diagnostic can be silenced.

## Excess Number of Function Arguments
crates/rstest-bdd-server/tests/smoke_lsp/wire.rs: shutdown_and_exit

### What lead to degradation?
shutdown_and_exit has 5 arguments, max arguments = 4

### Why does this problem occur?
Functions with many arguments indicate either a) low cohesion where the function has too many responsibilities, or b) a missing abstraction that encapsulates those arguments.
The threshold for the Rust language is 4 function arguments.

### How to fix it?
Start by investigating the responsibilities of the function. Make sure it doesn't do too many things, in which case it should be split into smaller and more cohesive functions. Consider the refactoring [INTRODUCE PARAMETER OBJECT](https://refactoring.com/catalog/introduceParameterObject.html) to encapsulate arguments that refer to the same logical concept.

### Helpful refactoring examples
This code health issue has been solved before in this project. Here are some examples for inspiration:

#### undefined

~~~diff
# diagnostics_fixture.rs (c22b439)
 #[cfg(feature = "diagnostics")]
-fn record_skipped_with_bypass(
-    scenario_name: &str,
+struct SkippedScenarioFixture<'a> {
+    scenario_name: &'a str,
     line: u32,
     tags: Vec<String>,
-    message: &str,
+    message: &'a str,
     allow_skipped: bool,
     forced_failure: bool,
+}
+
+#[cfg(feature = "diagnostics")]
+fn record_skipped_with_bypass(
+    scenario: SkippedScenarioFixture<'_>,
     bypassed_step: (StepKeyword, &str),
 
+    let SkippedScenarioFixture {
+        scenario_name,
+        line,
+        tags,
+        message,
+        allow_skipped,
+        forced_failure,
+    } = scenario;
+
     let metadata =
     bdd::record_bypassed_steps(
-        FEATURE_PATH,
-        scenario_name,
-        line,
-        tags,
-        Some(message),
+        bdd::BypassedScenario::new(FEATURE_PATH, scenario_name, line)
+            .with_tags(&tags)
+            .with_reason(Some(message)),
         [bypassed_step],
     record_skipped_with_bypass(
-        "fixture skipped scenario",
-        7,
-        vec!["@allow_skipped".into()],
-        "fixture skip message",
-        true,
-        false,
+        SkippedScenarioFixture {
+            scenario_name: "fixture skipped scenario",
+            line: 7,
+            tags: vec!["@allow_skipped".into()],
+            message: "fixture skip message",
+            allow_skipped: true,
+            forced_failure: false,
+        },
         (StepKeyword::Given, "fixture bypassed step"),
     record_skipped_with_bypass(
-        "fixture forced failure skip",
-        12,
-        vec!["@critical".into()],
-        "fixture forced skip",
-        false,
-        true,
+        SkippedScenarioFixture {
+            scenario_name: "fixture forced failure skip",
+            line: 12,
+            tags: vec!["@critical".into()],
+            message: "fixture forced skip",
+            allow_skipped: false,
+            forced_failure: true,
+        },
         (StepKeyword::Then, "fixture forced bypass"),
~~~

File: Makefile [hunk 2]
```diff
@@ -215,3 +215,14 @@ vale: ## Check prose
...
 	$(UV) run $(ACRONYM_SCRIPT)
 	$(VALE) --no-global --output line .
+
+# Opt-in accelerated debug builds (Cranelift + mold); requires a nightly
+# toolchain. See AGENTS.md and tools/dev-fast/config.toml.
+DEV_FAST_CONFIG ?= tools/dev-fast/config.toml
+
+.PHONY: dev-build dev-test
+dev-build: ## Build debug binaries with Cranelift and mold
+	cargo --config "$(DEV_FAST_CONFIG)" build
```
[the remark this answers is not in the document]
Comment (leynos) [reply Makefile]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

Do not treat warnings as optional or aspirational. Where a change is out of scope for this PR, propose a GitHub issue unless one exists already. (Treat o11y, code safety, documentation and validation coverage as in scope).

### ❌ Failed checks (1 error, 2 warnings)

| Check name | Status | Explanation | Resolution |
| :---: | :--- | :--- | :--- |
| Testing (Overall) | ❌ Error | The pull request adds behavioural Makefile functionality without substantive tests. The diff adds `dev-build` and `dev-test`, which invoke Cargo with `tools/dev-fast/config.toml`; the configuration se… | Add substantive integration tests for the new Make targets and configuration. Use a controlled fake `cargo` executable or an isolated fixture so the tests verify that both targets pass the configured path and the expected Cargo subcommands.… |
| Developer Documentation | ⚠️ Warning | The PR adds `make dev-build` and `make dev-test`, the `tools/dev-fast/config.toml` Cargo configuration, the Cranelift backend, and the Linux `mold` prerequisite. The guide requires a nightly toolchain… | Add a section to `docs/developers-guide.md` that documents `make dev-build` and `make dev-test`, `DEV_FAST_CONFIG`, the explicit `--config` usage, the nightly toolchain requirement, the Linux `mold`/`PATH` requirement, Cranelift's debug-bui… |
| Testing (Unit And Behavioural) | ⚠️ Warning | The pull request adds two externally observable command-line workflows, `make dev-build` and `make dev-test`, but adds no unit, behavioural, or end-to-end tests. The diff contains only configuration, … | Add end-to-end integration tests at the Makefile command boundary. Run both targets with the supported nightly toolchain and Linux `mold` setup, and assert that the explicit configuration enables Cranelift and the Linux linker flag. Cover c… |
